webpack/components/extensions/RegistrationCommands/fields/ActivationKeys.js

```diff
@@ -38,10 +38,16 @@ const ActivationKeys = ({
   };
 
   // Validate field when hostgroup is changed (host group may have some keys)
-  useEffect(() => {
+  useEffect(() => { 
     handleInvalidField('Activation Keys', akHasValidValue(hostGroupId, pluginValues?.activationKeys, hostGroupActivationKeys));
   }, [handleInvalidField, hostGroupId, hostGroupActivationKeys, pluginValues]);
 
+  useEffect(() => {
+    if (activationKeys?.length === 1) {
+      updatePluginValues([activationKeys[0]?.name]);
```

Comment:
If `activationKeys` is not empty, we can safely assume that the key has a name attribute:
`updatePluginValues([activationKeys[0].name]);`